### PR TITLE
Add Remote Config Integration tests

### PR DESCRIPTION
**Pull request by lahirumaramba**
- Add Remote Config Integration tests

Note: Added `release:stage` to trigger integration tests

Related to: #446

### CODE REVIEW

File: src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    assertEquals(String.format("Rollback to version %s", versionNumber),
+            rolledBackTemplate.getVersion().getDescription());
+
+    // get template to verify rollback
+    Template activeTemplate = remoteConfig.getTemplate();
+    assertEquals(rolledBackTemplate, activeTemplate);
+  }
+
+  @Test
+  public void testListVersions() throws Exception {
```
Comment (hiranya911) [inline src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java:131]: I'd suggest using the sync APIs here like the previous test case.

File: src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+  private static final Version VERSION = Version
+          .withDescription(String.format("promo config %s", timestamp));
+
+  @BeforeClass
+  public static void setUpClass() {
+    remoteConfig = FirebaseRemoteConfig.getInstance(IntegrationTestUtils.ensureDefaultApp());
+  }
+
+  @Test
+  public void testTemplateOperations() throws FirebaseRemoteConfigException {
```
Comment (hiranya911) [inline src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java:83]: Nit: Some of these can be split into separate test case with a bit of effort and extra RPCs (I'll leave it to you to decide whether that's worth it)

1. validate can be easily turned into a separate test case
2. getAtVersion and rollback can also be made into separate test cases if we are willing to run a few extra get and publish operations.

**Review by hiranya911 — APPROVED** [review]
LGTM with some suggestions.

File: src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java (hunk 1 above)
Comment (lahirumaramba) [reply src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java:131]: Updated to use the sync api

File: src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java (hunk 2 above)
Comment (lahirumaramba) [reply src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigIT.java:83]: Good point! As discussed offline let's address this in a future PR. Thanks!

**Pull request merged**